src/encord_active/lib/encord/actions.py

```diff
@@ -157,71 +225,74 @@ def create_project(
 
-                initiated_label_row["object_answers"] = original_label_row["object_answers"]
-                initiated_label_row["classification_answers"] = original_label_row["classification_answers"]
+        return new_project
 
-                # Remove unused object/classification answers
-                for object_hash in object_hashes:
-                    initiated_label_row["object_answers"].pop(object_hash)
+    def fix_pickle_files(self, embedding_type: EmbeddingType, renaming_map: dict[str, str]):
+        def fix_pickle_file(embedding: LabelEmbedding, renaming_map: dict[str, str]):
+            embedding["label_row"] = renaming_map[embedding["label_row"]]
+            embedding["data_unit"] = renaming_map[embedding["data_unit"]]
+            url_without_extension, extension = embedding["url"].split(".")
+            changed_parts = [renaming_map[x] if x in renaming_map else x for x in url_without_extension.split("/")]
+            embedding["url"] = "/".join(changed_parts) + "." + extension
+            return embedding
+
+        collection = load_collections(embedding_type, self.project_file_structure.embeddings)
+        updated_collection = [fix_pickle_file(up, renaming_map) for up in collection]
+        save_collections(embedding_type, self.project_file_structure.embeddings, updated_collection)
+
+    def replace_uids(
+        self, file_mappings: dict[LabelRowDataUnit, LabelRowDataUnit], project_hash: str, dataset_hash: str
+    ):
+        label_row_meta = json.loads(self.project_file_structure.label_row_meta.read_text(encoding="utf-8"))
+        original_dataset_hash = next(iter(label_row_meta.values()))["dataset_hash"]
 
-                for classification_hash in classification_hashes:
-                    initiated_label_row["classification_answers"].pop(classification_hash)
+        renaming_map = {self.project_meta["project_hash"]: project_hash, original_dataset_hash: dataset_hash}
 
-                initiated_label_row = construct_answer_dictionaries(initiated_label_row)
-                new_project.save_label_row(initiated_label_row["label_hash"], initiated_label_row)
+        for (old_lr, old_du), (new_lr, new_du) in file_mappings.items():
+            old_lr_path = self.project_file_structure.data / old_lr
+            (old_lr_path / "images" / f"{old_du}.jpg").rename(old_lr_path / "images" / f"{new_du}.jpg")
+            renaming_map[old_lr], renaming_map[old_du] = new_lr, new_du
 
-                # remove unused object and classification answers
+        dir_renames = {old_lr: new_lr for (old_lr, old_du), (new_lr, new_du) in file_mappings.items()}
+        for (old_lr, new_lr) in dir_renames.items():
+            (self.project_file_structure.data / old_lr).rename(self.project_file_structure.data / new_lr)
 
-            if progress_callback:
-                progress_callback((counter + 1) / len(all_new_label_rows))
+        rename_identifiers(renaming_map)
```

Comment:
[nitpick] 💅 

Don't think this helper method is providing any value, on the contrary, it hides the fact this access the DB.

```suggestion
        MergedMetrics().replace_identifiers(renaming_map)
```